Modules/Core/Common/src/itkOutputWindow.cxx

```diff
@@ -28,13 +28,15 @@
 #include "itkOutputWindow.h"
 #include "itkObjectFactory.h"
 #include "itkSingleton.h"
+#include <mutex>
 
 namespace itk
 {
 
 struct OutputWindowGlobals
 {
   OutputWindow::Pointer m_Instance{ nullptr };
+  std::recursive_mutex  m_StaticInstanceLock;
```

Comment:
Interesting! For my understanding, can you explain why `std::recursive_mutex` was chosen here, instead of the more lightweight `std::mutex`?

Reply:
That was the pattern used in the other "Instances". The global pimple pattern used is so complex that I am not sure any guarantees can be made that the recursive mutex will not be required.

Reply:
Thanks for explaining @blowekamp